lib/router/router.js

```diff
@@ -260,7 +260,12 @@ export default class Router extends EventEmitter {
     const jsonPageRes = await this.fetchRoute(route)
     const jsonData = await jsonPageRes.json()
 
-    if (jsonData.buildIdMismatch) {
+    const { buildId } = window.__NEXT_DATA__
+
+    if (
+      process.env.NODE_ENV === 'production' &&
+      buildId !== jsonPageRes.headers.get('Next-Build-Id')
```

Comment:
I tested with few options.
@rauchg here we've two options.

#### Option 1
We read the JSON page at runtime, add buildId to the JSON and send.
Now, we need to write our own JSON sending logic and we can't use npm `send` module.
Also, we may need to load the JSON page to to RAM which increase the RAM usage and latency.

#### Option 2
Add buildId when writing the JSON page.
Now we need to build the BUILD_ID before doing the webpack compilation.
And the buildId is hardcoded on every JSON page.

### Problems

#### Option 1
We want to write our own logic of the `send` module and it might increase our RAM usage.

#### Option 2
Now it's hard to change to support a custom BUILD_ID. Earlier it's simply as change `.next/BUILD_ID`. This is useful is build environment where you build the app in each instance. See: https://github.com/zeit/next.js/issues/786

----

I'm in favor of option 2 and add support to an official way to set a custom BUILD_ID.
What do you think?